src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -2528,8 +2532,33 @@ bool RTPSParticipantImpl::is_reader_ignored(
 }
 
 bool RTPSParticipantImpl::ignore_participant(
-        const GuidPrefix_t& /*participant_guid*/)
+        const GuidPrefix_t& participant_guid)
 {
+    {
```

Comment:
We should also avoid ignoring ourselves.

```suggestion
    if (participant_guid == m_guid.guidPrefix)
    {
        return false;
    }
    
    {
```